cloudinit/sources/DataSourceCloudCIX.py

```diff
@@ -0,0 +1,206 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import json
+import logging
+from typing import Optional
+
+from cloudinit import dmi, net, sources, url_helper, util
+from cloudinit.net.dhcp import NoDHCPLeaseError
+from cloudinit.net.ephemeral import EphemeralIPNetwork
+
+LOG = logging.getLogger(__name__)
+
+METADATA_URLS = ["http://169.254.169.254"]
+METADATA_VERSION = 1
+
+BUILTIN_DS_CONFIG = {
+    "retries": 3,
+    "timeout": 5,
+    "wait": 5,
+}
```

Comment:
It's probably best to define these as class attributes, and avoid trying to set these up in __init__ as there is generalized logic in DataSource.get_url_params to source `url_*` class attrs and fallback to ds_cfg settings if present.

Reply:
Done.